src/app/backend/resource/deployment/rollout.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2017 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package deployment
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	v1 "k8s.io/api/apps/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	client "k8s.io/client-go/kubernetes"
+)
+
+// RolloutSpec is a specification for deployment rollout
+type RolloutSpec struct {
+	// Revision is the requested/resulted revision number of the ReplicateSet to rollback.
+	Revision string `json:"revision"`
+}
+
+// RollbackDeployment rollback to a specific ReplicaSet revision
+func RollbackDeployment(client client.Interface, rolloutSpec *RolloutSpec, namespace, name string) (*RolloutSpec, error) {
+	deployment, err := client.AppsV1().Deployments(namespace).Get(context.TODO(), name, metaV1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	currRevision := deployment.Annotations["deployment.kubernetes.io/revision"]
```

Comment:
Export annotation string and `1` version string to const values. I'd put them on top.

Reply:
done